src/app/shared/palette/order-summary/read-order-summary-field.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, ViewChild, ReflectiveInjector,
+  ComponentFactoryResolver, ViewContainerRef, ApplicationRef, Injector, Type, ViewChildren, QueryList, OnInit } from '@angular/core';
+import { AbstractFieldReadComponent } from '../base-field/abstract-field-read.component';
+import { PaletteService } from '../palette.service';
+import { ReadMoneyGbpFieldComponent } from '../money-gbp/read-money-gbp-field.component';
+import { Fee } from './fee.model';
+import { CaseField } from '../../domain/definition/case-field.model';
+import { ReadTextFieldComponent } from '../text/read-text-field.component';
+import { ReadOrderSummaryRowComponent } from './read-order-summary-row.component';
+import { FeeValue } from './fee-value.model';
+import { MoneyGBPCaseField } from '../../domain/definition/money-gbp-case-field.model';
+
+@Component({
+  selector: 'ccd-read-order-summary-field',
+  templateUrl: './read-order-summary-field.html',
+  styleUrls: [
+    './read-order-summary-field.scss'
+  ],
+})
+export class ReadOrderSummaryFieldComponent extends AbstractFieldReadComponent implements OnInit {
+
+  public static readonly NO_VALUE = undefined;
+  @ViewChild('paymentTotal', {read: ViewContainerRef})
+  paymentTotal: ViewContainerRef;
+  @ViewChildren(ReadOrderSummaryRowComponent)
+  orderSummaryRows: QueryList<ReadOrderSummaryRowComponent>;
+
+  constructor(
+    private resolver: ComponentFactoryResolver,
+    private appRef: ApplicationRef,
+    private injector: Injector
+  ) {
+    super();
+  }
+
+  ngOnInit(): void {
+    let resolvedInputs = ReflectiveInjector.resolve([]);
+    let injector = ReflectiveInjector.fromResolvedProviders(resolvedInputs);
+    let paymentTotalComponent = this.resolver.resolveComponentFactory(ReadMoneyGbpFieldComponent).create(injector);
+
+    paymentTotalComponent.instance['caseField'] = this.caseField.value ?
+        new MoneyGBPCaseField(this.caseField.value.PaymentTotal)
+      : new MoneyGBPCaseField(ReadOrderSummaryFieldComponent.NO_VALUE);
```

Comment:
can we make the amount parameter optional and call:
`new MoneyGBPCaseField()`

Reply:
Do we ever consider undefined value a valid instance of money gbp?

Reply:
to me it's partially complete state which we should never issue as such